### PR TITLE
Add use case/example for oc rsync

### PR DESCRIPTION
**Pull request by sallyom**

### CODE REVIEW

**Comment by sallyom** [comment]
@danmcp @bparees ptal, thanks

**Comment by bparees** [comment]
I think @danmcp was looking for a full example of copying a db archive into a pod and then rsh'ing into the pod and restoring it. Though that's going to have a lot of overlap w/ what's going into the v2 migration guide.

File: dev_guide/copy_files_to_container.adoc [hunk 1]
```diff
@@ -12,7 +12,7 @@ toc::[]
 
 == Overview
 
-You can use the CLI to copy local files to or from a remote directory in a container.  This is a useful tool for copying database archives to and from your pods for backup and restore purposes.  It can also be used to copy source code changes into a running pod for development debugging, when the running pod supports hot reload of source files.
+You can use the CLI to copy local files to or from a remote directory in a container.  This is a useful tool for copying database archives to and from your pods for backup and restore purposes.  Note that a database pod's persistent volume directory is 'var/lib/[mysql|mongodb|pgsql]/data'.  This tool can also be used to copy source code changes into a running pod for development debugging, when the running pod supports hot reload of source files.
```
Comment (bparees) [inline dev_guide/copy_files_to_container.adoc]: path should be absolute (/var/lib/......) but i'm not sure we want to specify this here anyway.  There could be other database images that use other paths, this only covers the images we provide.  Doc about what path specific images use for volumes belongs w/ the image specific documentation.

**Comment by danmcp** [comment]
> I think @danmcp was looking for a full example of copying a db archive into a pod and then rsh'ing into the pod and restoring it. Though that's going to have a lot of overlap w/ what's going into the v2 migration guide.

@bparees @sallyom 
I am looking for something close to that.  But export from a running v3 container and import to a running v3 container.  Even more generically I am looking for some pointers on how to work with running database containers.  Tuning (like running statistics/analyze) would be another example we could show.

File: dev_guide/copy_files_to_container.adoc [hunk 2]
```diff
@@ -40,6 +40,44 @@ $ oc rsync devpod1234:/src /home/user/source
 ----
 ====
 
+[[copying-usage-with-database-restore]]
+== Usage With Database Restore
+
+Use `oc rsync` to copy local database archives to a database pod's persistent volume directory.
+Note: MySQL is used here, but replace 'mysql|MYSQL' with 'pgsql|PGSQL' or 'mongodb|MONGODB' to use
```
Comment (ahardin-rh) [inline dev_guide/copy_files_to_container.adoc]: Let's place this note in a note box

[NOTE]
====
MySQL is used here, but replace 'mysql|MYSQL' with 'pgsql|PGSQL' or 'mongodb|MONGODB' to use any of the three supported databases. 
====

File: dev_guide/copy_files_to_container.adoc [hunk 3]
```diff
@@ -40,6 +40,44 @@ $ oc rsync devpod1234:/src /home/user/source
...
+any of the three supported databases.  For this example, files are copied to database pod's
+'/var/lib/<db-service>/data' directory:
```
Comment (ahardin-rh) [inline dev_guide/copy_files_to_container.adoc]: Replace '/var/lib/<db-service>/data' with  *_/var/lib/<db-service>/data_*

**Comment by ahardin-rh** [comment]
@sallyom @danmcp @bparees picking this back up. Made a few style comments, but from the above comments it looks like you're still fleshing this out.

**Comment by bparees** [comment]
@sallyom based on @danmcp's comments:

> I am looking for something close to that. But export from a running v3 container and import to a running v3 container. 

can you update this to include exporting from the db pod, then restoring to a different db pod?

> Even more generically I am looking for some pointers on how to work with running database containers. Tuning (like running statistics/analyze) would be another example we could show.

i think this can be a separate PR/item.

**Comment by sallyom** [comment]
@bparees, sure, I'm hoping to get to this by the end of the day, if not first thing Tues, thanks for clarifying the comment above.  @ahardin-rh thanks for the style review, I'll get it fixed up.

**Comment by sallyom** [comment]
@bparees, I updated to include the export of (already existing) archive files to local /tmp and import from local /tmp.  Since each db service has different backup/restore cmds, we could omit the restore cmds and link to the migration doc instead for the db-specific cmds.  Should we hold off on this PR until then or are the specific restore cmds useful here?

**Comment by bparees** [comment]
hm. you're also already omitting the backup commands.

I suggest you simplify this down to one good example that shows the full flow (oc rsh to old pod, run backup commands, oc rsync to local, oc rsync to new pod, oc rsh to new pod, run restore commands) for one DB system.  And then reference the migration doc like "to see exact commands for all our supported images, see the migration doc" (of course that means this PR can't merge until the migration doc is merged)

**Comment by sallyom** [comment]
@bparees, finally got back to this PR.  I added one mysql example that shows:
 - link to migration doc to find exact db backup/restore cmds
 - backup cmd for mysql 
 - rsync archive file to local from existing running db pod
 - oc new-app 2nd mysql pod w/ unique service name and persistent storage
 - rsync archive file from local to running 2nd mysql pod
 - restore db using mysql source command into the new persistent mysql pod

@ahardin-rh please review, not sure if I referenced the migration doc link correctly.  thanks!

File: dev_guide/copy_files_to_container.adoc [hunk 4]
```diff
@@ -41,6 +41,56 @@ $ oc rsync devpod1234:/src /home/user/source
 ----
 ====
 
+[[copying-usage-with-database-restore]]
```
Comment (bparees) [inline dev_guide/copy_files_to_container.adoc]: match the anchor text w/ the actual heading text

File: dev_guide/copy_files_to_container.adoc [hunk 5]
```diff
@@ -41,6 +41,56 @@ $ oc rsync devpod1234:/src /home/user/source
...
+== Usage With Database Restore
```
Comment (bparees) [inline dev_guide/copy_files_to_container.adoc]: change title to "Backing up and Restoring Databases"

**Review by bparees — CHANGES_REQUESTED** [review]
couple nits and lgtm.

File: dev_guide/copy_files_to_container.adoc (hunk 5 above)
Comment (sallyom) [inline dev_guide/copy_files_to_container.adoc]: changed, thanks

File: dev_guide/copy_files_to_container.adoc (hunk 4 above)
Comment (sallyom) [inline dev_guide/copy_files_to_container.adoc]: changed, thanks

**Comment by sallyom** [comment]
@ahardin-rh, ready for any style edits you have :) thanks

**Comment by ahardin-rh** [comment]
Work continued in https://github.com/openshift/openshift-docs/pull/3022

**Pull request closed without merging**